src/kbmod/work_unit.py

```diff
@@ -0,0 +1,170 @@
+import math
+
+from astropy.io import fits
+from astropy.table import Table
+import numpy as np
+from pathlib import Path
+
+from kbmod.configuration import SearchConfiguration
+from kbmod.search import ImageStack, LayeredImage, PSF, RawImage
+
+
+class WorkUnit:
+    """The work unit is a storage and I/O class for all of the data
+    needed for a full run of KBMOD, including the: the search parameters,
+    data files, and the data provenance metadata.
+
+    A WorkUnit file is a FITS file with the following extensions:
+        0 - Primary header with overall metadata
+        1 - The data provenance metadata
+        2 - The search parameters
+        3 through X - Image layers with either alternating layers of science, variance,
+            mask, and PSF. Layers may be empty if no data is provided.
+    """
+
+    def __init__(self, im_stack=None, config=None):
+        self.im_stack = im_stack
+        self.config = config
+
+    @classmethod
+    def from_file(cls, filename):
+        """Create a WorkUnit from a single file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The file to load.
+
+        Returns
+        -------
+        result : `WorkUnit`
+            The loaded WorkUnit.
+        """
+        if not Path(filename).is_file():
+            raise ValueError(f"WorkUnit file {filename} not found.")
+
+        result = None
+        with fits.open(filename) as hdul:
+            num_layers = len(hdul)
+            if num_layers < 5:
+                raise ValueError(f"WorkUnit file has too few extensions {len(hdul)}.")
+
+            # TODO - Read in provenance metadata from extension #1.
+
+            # Read in the search parameters from extension #2.
+            config = SearchConfiguration()
+            config_table = Table(hdul[2].data)
```

Comment:
I was kind of wondering earlier why you wouldn't just write the config as a header, is there a variable we have that breaks the length limit or something?